Sources/Subs-Db-mysql.php

```diff
@@ -1024,4 +1025,29 @@ function smf_db_native_replace()
 	return true;
 }
 
+/**
+ * Function which return the information if the database supports cte with recursive
+ *
+ * @return boolean true or false
+ */
+function smf_db_cte_support()
+{
+	global $smcFunc;
+	static $return;
+
+	if (!empty($return))
+		return $return;
+
```

Comment:
Nit picking this, but if $return is false because support doesn't exist, then this check is always ignored and the efficiency is lost.